File: medplat/medplat-web/src/main/java/com/argusoft/medplat/systemconstraint/service/SystemConstraintService.java
package com.argusoft.medplat.systemconstraint.service;

import com.argusoft.medplat.systemconstraint.dto.SystemConstraintFieldMasterDto;
import com.argusoft.medplat.systemconstraint.dto.SystemConstraintFormMasterDto;
import com.argusoft.medplat.systemconstraint.dto.SystemConstraintMobileTemplateConfigDto;
import com.argusoft.medplat.systemconstraint.dto.SystemConstraintStandardFieldMasterDto;
import com.argusoft.medplat.systemconstraint.model.SystemConstraintStandardFieldMaster;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface SystemConstraintService {

    List<SystemConstraintFormMasterDto> getSystemConstraintForms(Integer menuConfigId);

    String getSystemConstraintFormConfigByUuid(UUID uuid, String appName);

    List<SystemConstraintFieldMasterDto> getSystemConstraintFieldsByFormMasterUuid(UUID formMasterUuid);

    SystemConstraintFormMasterDto getSystemConstraintFormByUuid(UUID uuid);

    SystemConstraintFormMasterDto createOrUpdateSystemConstraintForm(SystemConstraintFormMasterDto systemConstraintFormMasterDto,String type);

    String createOrUpdateSystemConstraintFieldConfig(UUID formMasterUuid, SystemConstraintFieldMasterDto systemConstraintFieldMasterDto);

    void deleteSystemConstraintFieldConfig(UUID uuid);

    Map<String, Object> getSystemConstraintWebTemplateConfigsByMenuConfigId(Integer menuConfigId);

    List<SystemConstraintStandardFieldMasterDto> getSystemConstraintStandardFields(Boolean fetchOnlyActive);

    SystemConstraintStandardFieldMaster getSystemConstraintStandardFieldByUuid(UUID uuid);

    String getSystemConstraintStandardConfigById(Integer id);

    void createOrUpdateSystemConstraintStandardField(SystemConstraintStandardFieldMasterDto systemConstraintStandardFieldMasterDto);

    String createOrUpdateSystemConstraintStandardFieldConfig(SystemConstraintStandardFieldMasterDto systemConstraintStandardFieldMasterDto);

    List<SystemConstraintMobileTemplateConfigDto> getMobileTemplateConfig(String formCode);

    List<String> getActiveMobileForms();

    String getSystemConstraintConfigsByMenuConfigId(Integer menuConfigId);
}

File: medplat/medplat-ui/app/admin/applicationmanagement/managesystemconstraints/services/system-constraint.service.js
(function () {
    function SystemConstraintService($resource, APP_CONFIG) {
        let api = $resource(APP_CONFIG.apiPath + '/systemConstraint/:action/:subAction/:id', {}, {
            getSystemConstraintForms: {
                method: 'GET',
                isArray: true
            },
            getSystemConstraintFormByUuid: {
                method: 'GET'
            },
            getSystemConstraintFormConfigByUuid: {
                method: 'GET'
            },
            getSystemConstraintFieldsByFormMasterUuid: {
                method: 'GET',
                isArray: true
            },
            getSystemConstraintStandardFields: {
                method: 'GET',
                isArray: true
            },
            getSystemConstraintStandardFieldByUuid: {
                method: 'GET'
            },
            createOrUpdateSystemConstraintForm: {
                method: 'POST'
            },
            createOrUpdateSystemConstraintFieldConfig: {
                method: 'POST'
            },
            deleteSystemConstraintFieldConfig: {
                method: 'DELETE'
            },
            getSystemConstraintStandardConfigById: {
                method: 'GET'
            },
            createOrUpdateSystemConstraintStandardFieldConfig: {
                method: 'POST'
            },
            createOrUpdateSystemConstraintStandardField: {
                method: 'POST'
            },
        });
        return {
            getSystemConstraintForms: function () {
                return api.getSystemConstraintForms({
                    action: 'forms'
                }, {}).$promise;
            },
            getSystemConstraintFormsByMenuConfigId: function (menuConfigId) {
                return api.getSystemConstraintForms({
                    action: 'forms',
                    subAction: menuConfigId
                }, {}).$promise;
            },
            getSystemConstraintFormByUuid: function (uuid) {
                return api.getSystemConstraintFormByUuid({
                    action: 'form',
                    subAction: uuid
                }, {}).$promise;
            },
            getSystemConstraintFormConfigByUuid: function (uuid, appName) {
                return api.getSystemConstraintFormConfigByUuid({
                    action: 'formConfig',
                    subAction: uuid,
                    appName: appName
                }, {}).$promise;
            },
            getSystemConstraintFieldsByFormMasterUuid: function (formMasterUuid) {
                return api.getSystemConstraintFieldsByFormMasterUuid({
                    action: 'fields',
                    subAction: formMasterUuid
                }, {}).$promise;
            },
            getSystemConstraintStandardFields: function (fetchOnlyActive = false) {
                return api.getSystemConstraintStandardFields({
                    action: 'standard',
                    subAction: 'fields',
                    fetchOnlyActive
                }, {}).$promise;
            },
            getSystemConstraintStandardFieldByUuid: function (uuid) {
                return api.getSystemConstraintStandardFieldByUuid({
                    action: 'standard',
                    subAction: 'field',
                    id: uuid
                }, {}).$promise;
            },
            createOrUpdateSystemConstraintForm: function (dto, type) {
                return api.createOrUpdateSystemConstraintForm({
                    action: 'form',
                    id: type
                }, dto).$promise;
            },
            createOrUpdateSystemConstraintFieldConfig: function (formMasterUuid, dto) {
                return api.createOrUpdateSystemConstraintFieldConfig({
                    action: 'fieldConfig',
                    subAction: formMasterUuid
                }, dto).$promise;
            },
            deleteSystemConstraintFieldConfig: function (uuid) {
                return api.deleteSystemConstraintFieldConfig({
                    action: 'fieldConfig',
                    subAction: uuid
                }, {}).$promise;
            },
            getSystemConstraintStandardConfigById: function (id) {
                return api.getSystemConstraintStandardConfigById({
                    action: 'standardConfig',
                    subAction: id
                }, {}).$promise;
            },
            createOrUpdateSystemConstraintStandardFieldConfig: function (dto) {
                return api.createOrUpdateSystemConstraintStandardFieldConfig({
                    action: 'standard',
                    subAction: 'fieldConfig'
                }, dto).$promise;
            },
            createOrUpdateSystemConstraintStandardField: function (dto) {
                return api.createOrUpdateSystemConstraintStandardField({
                    action: 'standard',
                    subAction: 'field'
                }, dto).$promise;
            },
        };
    }
    angular.module('imtecho.service')
        .factory('SystemConstraintService', ['$resource', 'APP_CONFIG', SystemConstraintService]);
})();
